torch/distributed/rpc/internal.py

```diff
@@ -136,5 +146,31 @@ def _load_return_value(binary_data, tensor_table):
     return result
 
 
+def _start_record_function(exec_type, func_name, current_worker_name, dest_worker_name):
+    """
+    This function should be called from RPC/RRef functions to create a
+    RecordFunction object for profiling.
+
+    Arguments:
+        exec_type (RPCExecMode): Type of RPC/RRef call
+        func_name (str): Name of function being profiled.
+        current_worker_name (str): Name of current worker.
+        dest_worker_name (str): Name of the destination worker.
+
+    Returns:
+        An instance of `torch.autograd._RecordFunction`. This function also has
+        the side effect of running the before callbacks that start the
```

Comment:
This is actually the main purpose of the function and not a side effect right? Maybe we should mention that in the main doc above.